### DISCUSSION TITLE
Modbus Network connection SUN2000-5KTL-M1

### DISCUSSION THREAD

**Discussion by Fettkeewl** [wlcrs/huawei_solar#228] [Setting up the integration] [answered]
Hello.

I'm currently running the other avaliable integration for Huawei Solar Inverter but it does not seem to be actively developed.

I was wondering if this integration supports the
SUN2000-5KTL-M1 and SUN2000-10KTL-M1?

I thought I'd ask before I completely venture into redoing every setting I have made for the other integration..

Currently communicating over home network, not the inverters own AP. Over port 502

**Comment by ruteclrp** [comment, marked as answer]
I am running the integration on  my SUN2000-8KTL-M1 via LAN connection. This is working out of the box - almost :-)

**Reply by Fettkeewl** [reply]
Are you running it over your Home LAN as in the same subnet as the rest of your devices and HA instance? :) ty for taking the time to reply

**Reply by ruteclrp** [reply]
Yes, everything runs off my router's DHCP with multiple switches here and there in between devices in my house, inclusive the inverter smartdongle FE.
I just plugged in the Raspberry PI4 directly at some empty port in the living room.

**Reply by Fettkeewl** [reply]
Ty, then I shall find time to migrate to this, I don't like to use abandoned software 😂

**Reply by alexualbu** [reply]
@ruteclrp , what firmware version do you have? and also did you enable Local q&M ?
I have V100R001C00SPC148 for the inverter and V100R001C00SPC133 for the dongle, only port 502 open, Modbus TCP enabled but the connection does not stick (it connects, then gets interrupted)

**Reply by ruteclrp** [reply]
same fw on inverter, but V100R001C00SPC130 on dongle. are you on LAN or wifi with dongle? I was specifically advised to use cabled LAN to inverter by my installer to avoid loosing the connection from time to time. I installed on port 502 just as in the instructions. Was never prompted for elevated permissions, but have full rights anyway. I am no on the latest beta version and have never had any issues. what do you mean by q&M ?

**Reply by alexualbu** [reply]
thank you for the reply!

i got to the bottom of it. i believe there is a bug with the configuration flow whereby the client used to validate the connection is not closed properly (because it continues to debug log even after the config window is closed) - turns out i was supposed to use slave_id 1, but because i first tried with 0 I couldn't complete the flow until restarting home assistant.

**Comment by Fettkeewl** [comment]
Transfer complete 👍
